src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -738,6 +741,9 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
                                 throw WinHttpException.CreateExceptionUsingError(lastError);
                             }
                         }
+
```

Comment:
If there's an error in any of this initialization, do we want to dispose of the session handle (if it's already being done and I'm just done seeing it in the diff, great) rather than leaving it for finalization?  Feel free to tell me it's not worth it.